portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that encapsulates the subtraction game. */
+@WebServlet("/comments")
+public final class CommentsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String numRequest = getParameter(request, "num", "");
+    int num = Integer.parseInt(numRequest);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String json = getDataStoreLimit(datastore, num);
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {}
+
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  private static String getDataStoreLimit(DatastoreService datastore, int num) {
```

Comment:
Let's pick a better name for this function. What does it actually return?
getDataStoreLimit makes me think it returns the limit of the data store as an int.

JavaDoc would also help clear things up for me.

Reply:
changed to getXNumOfComments. If my function name is proper, would javadoc be necessary?

Reply:
Javadoc isn't necessary if it's obvious, but it's good practice. Google's linter might even require Javadoc if there's a return (not sure on that).
I'm fine with the function name being just `getComments`. It's clear you're getting comments. You can also overload the function, have a `getComments` that only takes a datastore, and another that takes a limit as the 2nd argument. Then, it'll be pretty clear you can limit the number of comments returned.

With that said, I don't see your changes. Will wait for you to push your changes.

Reply:
changes pushes